test/NuGet.Core.Tests/NuGet.Packaging.Test/PackageDependencyGroupTests.cs

```diff
@@ -218,5 +222,108 @@ public void PackageDependencyGroup_GetHashCode_SameVersionAndPackages_DifferentO
 
             Assert.Equal(aHashCode, bHashCode);
         }
+
+        [Fact]
+        public void PackageDependencyGroup_Deserialize_ReturnsExpected()
+        {
+            // Arrange
+            const string id = "PackageA";
+            var version = new VersionRange(new NuGetVersion(1, 4, 1));
+            var expectedPackageDependency = new PackageDependency(id, version);
+            IEnumerable<PackageDependency> packages = new List<PackageDependency>() { expectedPackageDependency };
+
+            var targetFramework = NuGetFramework.Parse(".NETStandard1.0");
+
+            // Act
+            using var stringReader = new StringReader(PackageRegistrationDependencyGroupsJson);
+            using var jsonReader = new JsonTextReader(stringReader);
+            jsonReader.Read();
+
+            var serializer = JsonSerializer.Create(JsonExtensions.ObjectSerializationSettings);
+            PackageDependencyGroup actualPackageDependencies = serializer.Deserialize<PackageDependencyGroup>(jsonReader);
+
+            // Assert
+            Assert.Equal(packages, actualPackageDependencies.Packages);
+            Assert.Equal(targetFramework, actualPackageDependencies.TargetFramework);
+        }
+
+        [Fact]
+        public void PackageDependencyGroup_NoTargetFramework_Deserialize_ReturnsExpected()
+        {
+            // Arrange
+            const string id = "PackageA";
+            var version = new VersionRange(new NuGetVersion(1, 4, 1));
+            var expectedPackageDependency = new PackageDependency(id, version);
+            IEnumerable<PackageDependency> packages = new List<PackageDependency>() { expectedPackageDependency };
+
+            var targetFramework = NuGetFramework.AnyFramework;
+
+            // Act
+            using var stringReader = new StringReader(PackageRegistrationDependencyGroupsJson_NoTargetFramework);
+            using var jsonReader = new JsonTextReader(stringReader);
+            jsonReader.Read();
```

Comment:
Please help me understand why we have to invoke the Read method in the tests.

Reply:
It is because the first token is null and doing the `Read()` moves the iterator to the json object, I'm not 100% sure why this happens but looks like we are doing it in other places.